packages/firestore/src/util/input_validation.ts

```diff
@@ -166,7 +166,8 @@ export function tryGetCustomObjectType(input: object): string | null {
 }
 
 /**
- * Casts `obj` to `T`. Throws if  `obj` is not an instance of `T`.
+ * Casts `obj` to `T`, optionally unwrapping Compat types to expose the
```

Comment:
Can we just check `obj._delegate` instead of relying on `instanceof`ing the abstract class, so that I can put it into `@firebase/util` and share with other SDKs?

Reply:
Updated. This function also does a name check and provides a different error message if there it detects that two classes share the same name but are not compatible. I was wondering if we need to strip suffixes here, since Rollup adds "$1" and counting during the build phase?

Reply:
Thanks. Yes, rollup will append `$x` to things with the same name if they end up in the same bundle. I'm not sure if it's the case with webpack which creates many closures instead of one big top level scope. I think it very much depends on the format of the bundle, thus is hard for us to predict/optimize for. I would just leave it as is for now and see if developer reports that the error is not helpful.